File: 1200/2181H/main.cpp
#include <bits/stdc++.h>
using namespace std;

long long gcd(long long a, long long b) 
{
    if (b == 0) 
        return a;

    return gcd(b, a % b);
}

int main() 
{
    long long w, h, d, n;
    cin >> w >> h >> d >> n;

    long long cnt = n;
    long long a = gcd(w, cnt);
    cnt /= a;

    long long b = gcd(h, cnt);
    cnt /= b;

    long long c = gcd(d, cnt);
    cnt /= c;

    if (cnt != 1)
        cout << -1 << "\n";
    else
        cout << (a - 1) << " " << (b - 1) << " " << (c - 1) << "\n";

    return 0;
}

File: 1200/2181H/main.java
import java.util.*;
public class main
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        long w = sc.nextLong();
        long h = sc.nextLong();
        long d = sc.nextLong();
        long n = sc.nextLong();
        
        long cnt = n;
        long a = gcd(w,cnt);
        
        cnt /= a;
        long b = gcd(h,cnt);
        
        cnt /= b;
        long c = gcd(d,cnt);
        
        cnt /= c;
 
        if(cnt != 1)
        {
            System.out.println(-1);
            return;
        }
        else
            System.out.println((a-1) + " " + (b-1) + " " + (c-1));
 
 
    }
    private static long gcd(long a,long b)
    {
        if(b==0) 
            return a;
            
        return gcd(b, a%b);
    }
}